### ISSUE TITLE
EWS API ConvertID not supported / implemented

### ISSUE THREAD

**Issue by scheidleon** [grommunio/gromox#76]
The following EWS Request returns 500 

```
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Header>
    <t:RequestServerVersion Version="Exchange2007_SP1" />
    <t:TimeZoneContext>
      <t:TimeZoneDefinition Id="W. Europe Standard Time" />
    </t:TimeZoneContext>
  </soap:Header>
  <soap:Body>
    <m:ConvertId DestinationFormat="EwsId">
      <m:SourceIds>
        <t:AlternateId Format="HexEntryId" Id="000000008586831ABED9C14194C55D3D89017599010001000000A5187B6FBCDCEA1ED03C565700000000000F0000" Mailbox="example@testdomain.local" />
      </m:SourceIds>
    </m:ConvertId>
  </soap:Body>
</soap:Envelope>
```

Im trying to synchronise calendar entrys with an external application. It seems the first thing the application tries is to convert the folder id from "EwsID" to "HexEntryID" which immediatly fails on grommunio (HTTP 500)

The Response from grommunio:

```
<SOAP:Envelope xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/1999/XMLSchema-instance">
  <SOAP:Body>
    <SOAP:Fault>
      <faultcode xsi:type="xsd:string">SOAP:Server</faultcode>
      <faultstring xsi:type="xsd:string">Unknown request 'ConvertId'.</faultstring>
    </SOAP:Fault>
  </SOAP:Body>
</SOAP:Envelope>
```

**Comment by juliaschroeder** [comment]
Yes, `ConvertId` is not implemented yet, as we didn't observe it "in the wild"  until now. I will put it on the list, but since our main focus is still on Mail and Outlook for Mac it may take some weeks until we get there.   
Which application are you using? It would be very helpful if we could test this ourselves.

**Comment by scheidleon** [comment]
Im using Easyjob 6. It's a special software for the event industry and allows syncing callendar entrys and notifications with an exchange server. They offer a free version, but i'm also happy to assist with testing if that helps.

I've tried to modify the response of `ConvertId` to something expected and that results in an `FindItem` request which is also not (yet) implemented as seen in the 2023.11.1 release notes.

Just as reference my modified response:
```
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
               xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
               xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <soap:Header>
    <t:ServerVersionInfo MajorVersion="8" MinorVersion="1" 
                         MajorBuildNumber="191" MinorBuildNumber="0" 
                         Version="Exchange2010" 
                         xmlns:t="https://schemas.microsoft.com/exchange/services/2006/types" />
  </soap:Header>
  <soap:Body>
    <m:ConvertIdResponse>
      <m:ResponseMessages>
        <m:ConvertIdResponseMessage ResponseClass="Success">
          <m:ResponseCode>NoError</m:ResponseCode>
          <m:AlternateId xsi:type="t:AlternateIdType" Format="HexEntryId" Id="000000008586831ABED9C14194C55D3D89017599010001000000A5187B6FBCDCEA1ED03C565700000000000F0000" Mailbox="example@testdomain.local" />
        </m:ConvertIdResponseMessage>
      </m:ResponseMessages>
    </m:ConvertIdResponse>
  </soap:Body>
</soap:Envelope>
```

The next Request by EasyJob:

```
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages" xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Header>
    <t:RequestServerVersion Version="Exchange2007_SP1" />
    <t:TimeZoneContext>
      <t:TimeZoneDefinition Id="W. Europe Standard Time" />
    </t:TimeZoneContext>
  </soap:Header>
  <soap:Body>
    <m:FindItem Traversal="Shallow">
      <m:ItemShape>
        <t:BaseShape>AllProperties</t:BaseShape>
        <t:AdditionalProperties>
          <t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="IdJob" PropertyType="String" />
        </t:AdditionalProperties>
      </m:ItemShape>
      <m:IndexedPageItemView MaxEntriesReturned="1" Offset="0" BasePoint="Beginning" />
      <m:Restriction>
        <t:IsEqualTo>
          <t:ExtendedFieldURI DistinguishedPropertySetId="PublicStrings" PropertyName="IdJob" PropertyType="String" />
          <t:FieldURIOrConstant>
            <t:Constant Value="1" />
          </t:FieldURIOrConstant>
        </t:IsEqualTo>
      </m:Restriction>
      <m:ParentFolderIds>
        <t:FolderId Id="000000008586831ABED9C14194C55D3D89017599010001000000A5187B6FBCDCEA1ED03C565700000000000F0000" />
      </m:ParentFolderIds>
    </m:FindItem>
  </soap:Body>
</soap:Envelope>
```

Im currently also trying to get my compiled version of grommox running for testing purposes.
Feel free to contact me for any questions.

**Comment by grammmike** [comment]
> Im using Easyjob 6. It's a special software for the event industry and allows syncing callendar entrys and notifications with an exchange server. They offer a free version, but i'm also happy to assist with testing if that helps.
> 
> I've tried to modify the response of `ConvertId` to something expected and that results in an `FindItem` request which is also not (yet) implemented as seen in the 2023.11.1 release notes.

The ConvertId operation has been implemented with https://github.com/grommunio/gromox/commit/8414e7025e585e57e11765a4148cb8f3ebeb82a1 and received further fixes with https://github.com/grommunio/gromox/commit/1c9b3428add1caa10c95b3c5eb9263ea06eb56be
